  public T get() {
    final Errors errors = new Errors();
    try {
      T t = injector.callInContext(new ContextualCallable<T>() {
        public T call(InternalContext context) throws ResolveFailedException {
          InjectionPoint injectionPoint = context.getInjectionPoint();
          return internalFactory.get(errors, context, injectionPoint);
        }
      });
      errors.throwIfNecessary();
      return t;
    } catch (ResolveFailedException e) {
      throw errors.merge(e.getErrors()).toProvisionException();
    }
  }

